cogs/updates.py

```diff
@@ -1,38 +1,120 @@
+import logging
+from typing import Optional
+from datetime import datetime, timedelta, timezone
+
 import discord
 from discord.ext import commands
 
+from db.models import GuildData
+from utils.misc import ordinal
+from utils.checks import admin_command, staff_command, cogify
+from utils.converters import Index
+
+# updates operates by UTC dates
+UTC = timezone(offset=timedelta(), name='UTC')
+
+class UpdatesMessage(commands.Converter):
+    async def convert(self, ctx: commands.Context, message_id: str) -> discord.Message:
+        guild_data = await GuildData.get(guild_id = ctx.guild.id)
+        updates_channel = ctx.bot.get_channel(guild_data.updates_id)
+        update_message = await updates_channel.fetch_message(message_id)
+        return update_message
+
 
 class Updates(commands.Cog):
     def __init__(self, bot: commands.Bot) -> None:
         self.bot = bot
     
+    cog_check = cogify(staff_command())
+    
     @commands.hybrid_group()
     async def update(self, ctx: commands.Context):
         """
         Commands for handling updates.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        subcmds = []
+        for cmd in ctx.command.commands:
+            try:
+                usable = await cmd.can_run(ctx)
+                if usable:
+                    subcmds.append('`'+cmd.name+'`')
+            except commands.CommandError as e:
+                logging.debug(f'Command "{cmd.name}" check threw error, discarded in {ctx.command.name} group subcommand list.')
+        await ctx.send(f'Available subcommands: {", ".join(subcmds)}.')
     
     @update.command()
-    async def set(self, ctx: commands.Context):
+    @admin_command()
+    async def set(self, ctx: commands.Context, channel: Optional[discord.TextChannel] = None):
         """
         Set an updates channel.
-        """ 
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        """
+        if channel:
+            values = dict(updates_id = channel.id)
+            await GuildData.update_or_create(values, guild_id = ctx.guild.id)
+            await ctx.send(f'Updates channel set to <#{channel.id}> for this guild.')
+        else:
+            values = dict(updates_id = None)
+            await GuildData.update_or_create(values, guild_id = ctx.guild.id)
```

Comment:
Instead of a branch for this, I think a ternary operator such as `dict(updates_id = channel.id if channel else None)`, then you can combine with line 52 and 53. Then you'd only have an if used for the reply message.

Reply:
Done, thanks for the feedback!